### PR TITLE
feat(FEC-12364): Add support for GTM Plugin in V3 studio

### PR DESCRIPTION
**Pull request by JonathanTGold**
### Description of the Changes

Add support for GTM Plugin in V3 studio

related pr: https://github.com/kaltura/playkit-js-google-tag-manager/pull/1

Screen Shot 2022-07-13 at 12 59 05

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: app/js/controllers/gtmPluginCtrl.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+KMCMenu.controller('gtmPluginCtrl', ['$scope',
+	function ($scope) {
+		$scope.customEvents = '';
+
+		if(!("googleTagManager" in $scope.playerData.config.plugins)) {
+			$scope.playerData.config.plugins["googleTagManager"] = {disable: true};
+		}
+
+		// init radio buttons according to selected related entries source
```
Comment (SivanA-Kaltura) [inline app/js/controllers/gtmPluginCtrl.js:9]: update or remove comment

File: app/js/controllers/gtmPluginCtrl.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+		if ($scope.playerData.config.plugins["googleTagManager"].enabled) {
+			var data = $scope.playerData.config.plugins["googleTagManager"];
```
Comment (SivanA-Kaltura) [inline app/js/controllers/gtmPluginCtrl.js:11]: use let/const

File: app/js/directives.js [hunk 3]
```diff
@@ -31,7 +31,7 @@ DirectivesModule.directive('onFinishRender', ['$timeout','$compile', function ($
 						$('div:contains("AirPlay").panel').before('<div class="tabSubHeader AirPlay">Apple AirPlay</div><p class="categoryDesc">Use the below to enable casting capabilities from iOS devices, using the Apple AirPlay protocol.</p>');
 					}
 					if ($(".Analytics").length === 0){
-						$('div:contains("Youbora").panel').before('<div class="tabSubHeader Analytics">Analytics</div><p class="categoryDesc">Kaltura supports robust analytics via the Kaltura platform as well as via 3rd party analytics providers.</p>'); // add Analytics separator
+						$('div:contains("google Tag Manager").panel').before('<div class="tabSubHeader Analytics">Analytics</div><p class="categoryDesc">Kaltura supports robust analytics via the Kaltura platform as well as via 3rd party analytics providers.</p>'); // add Analytics separator
```
Comment (SivanA-Kaltura) [inline app/js/directives.js:34]: the div has no id/class ? and why is the google in lowercase ?

File: app/templates/googleTagManager.html [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div class="text" style="height: 115px" ng-controller="gtmPluginCtrl">
+    <span class="pluginLabel" ng-class="{disabled: plugin!=null && !plugin.enabled}" id="{{property.id}}">{{property.label}}</span>
```
Comment (SivanA-Kaltura) [inline app/templates/googleTagManager.html:2]: can plugin be undefined ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: app/js/controllers/gtmPluginCtrl.js (hunk 1 above)
Comment (JonathanTGold) [reply app/js/controllers/gtmPluginCtrl.js:9]: fixed

File: app/templates/googleTagManager.html (hunk 4 above)
Comment (JonathanTGold) [reply app/templates/googleTagManager.html:2]: this logic was copied from other templates I didn’t get into it

File: app/js/controllers/gtmPluginCtrl.js [hunk 5]
```diff
@@ -18,7 +18,9 @@ KMCMenu.controller('gtmPluginCtrl', ['$scope',
 		}
 
 		$scope.customListOfCustomEventsChange = function(){
-			$scope.playerData.config.plugins["googleTagManager"].customEventsTracking.custom = $scope.customListOfCustomEvents.split(',');
+			$scope.playerData.config.plugins["googleTagManager"].customEventsTracking.custom = $scope.customListOfCustomEvents
+				.replace(/\s/g,'')
```
Comment (giladna) [inline app/js/controllers/gtmPluginCtrl.js:22]: @JonathanTGold maybe \s+ ?

**Review by giladna — CHANGES_REQUESTED** [review]

File: app/js/controllers/gtmPluginCtrl.js (hunk 5 above)
Comment (JonathanTGold) [reply app/js/controllers/gtmPluginCtrl.js:22]: we don't need that, since we made a global search

**Review by giladna — APPROVED** [review]

**Review by giladna — APPROVED** [review]

**Pull request merged**